### PR TITLE
Add a tip about clearing parent routes

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is about docs.

### CODE REVIEW

File: Resources/doc/reference/architecture.rst [hunk 1]
```diff
@@ -356,5 +356,26 @@ Then, you have to set the CommentAdmin ``parentAssociationMapping`` attribute to
 
 It also possible to set a dot-separated value, like ``post.author``, if your parent and child admins are not directly related.
 
+Be wary that being a child admin is optional, which means that regular routes
+will be created regardless of whether you actually need them or not. To get rid
+of them, you may override the `configureRoute` method::
+
+    <?php
```
Comment (Soullivaneuh) [inline Resources/doc/reference/architecture.rst]: Missing `.. code-block:: php` before the block.

BTW, is the `<?php` opening tag really necessary?

File: Resources/doc/reference/architecture.rst [hunk 2]
```diff
@@ -356,5 +356,26 @@ Then, you have to set the CommentAdmin ``parentAssociationMapping`` attribute to
 
 It also possible to set a dot-separated value, like ``post.author``, if your parent and child admins are not directly related.
 
+Be wary that being a child admin is optional, which means that regular routes
+will be created regardless of whether you actually need them or not. To get rid
+of them, you may override the `configureRoute` method::
```
Comment (Soullivaneuh) [inline Resources/doc/reference/architecture.rst]: I don't think the double `:` is wanted here.

File: Resources/doc/reference/architecture.rst [hunk 3]
```diff
@@ -356,5 +356,26 @@ Then, you have to set the CommentAdmin ``parentAssociationMapping`` attribute to
 
 It also possible to set a dot-separated value, like ``post.author``, if your parent and child admins are not directly related.
 
+Be wary that being a child admin is optional, which means that regular routes
+will be created regardless of whether you actually need them or not. To get rid
```
Comment (Soullivaneuh) [inline Resources/doc/reference/architecture.rst]: Please don't break lines like this.

File: Resources/doc/reference/architecture.rst [hunk 4]
```diff
@@ -356,5 +356,26 @@ Then, you have to set the CommentAdmin ``parentAssociationMapping`` attribute to
...
+    class CommentAdmin extends AbstractAdmin
+    {
+        protected $parentAssociationMapping = 'post';
+
+        protected function configureRoutes(RouteCollection $collection)
+        {
+            if (!$this->isChild()) {
+                $collection->clear();
+            }
+        }
```
Comment (Soullivaneuh) [inline Resources/doc/reference/architecture.rst]: Maybe add an example with `clearExcept`?

File: Resources/doc/reference/architecture.rst (hunk 3 above)
Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: It's like that in the Symfony documentation standards, which we said we were following.

File: Resources/doc/reference/architecture.rst (hunk 2 above)
Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: It is, it's the standard notation, recommended by the Symfony documentation standards, which we said we were following.

File: Resources/doc/reference/architecture.rst (hunk 1 above)
Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: That's what the `::` is about. For the `<?php` I'm not sure, let me check

File: Resources/doc/reference/architecture.rst (hunk 3 above)
Comment (Soullivaneuh) [inline Resources/doc/reference/architecture.rst]: Are Symfony documention following this rule? Oo

Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: I don't know, but they say they want to : 

> Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);

File: Resources/doc/reference/architecture.rst (hunk 2 above)
Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: Syntactic coloration does not seem to work :(

Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: Got it to work, see second commit :)

File: Resources/doc/reference/architecture.rst (hunk 1 above)
Comment (greg0ire) [inline Resources/doc/reference/architecture.rst]: The `<?php` is required

**Comment by greg0ire** [comment]
There are 2 approvals, please merge.

**Comment by jordisala1991** [comment]
Thanks @greg0ire

**Pull request merged**